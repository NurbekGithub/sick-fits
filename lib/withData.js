import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpoint, prodEndpoint } from '../config';
import { LOCAL_STATE_QUERY } from '../components/Cart';

export function dump(data) {
  return JSON.stringify(data, null, 2);
}

function createClient({ headers }) {
  return new ApolloClient({
    uri: process.env.NODE_ENV === 'development' ? endpoint : prodEndpoint,
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include',
        },
        headers,
      });
    },
    clientState: {
      resolvers: {
        Mutation: {
          toggleCart(_, variables, { cache }) {
            // 1. read the cartOpen value from the cache
            const { cartOpen } = cache.readQuery({ query: LOCAL_STATE_QUERY });
            // 2. write the cartOpen value to the cache
            const data = {
              data: { cartOpen: !cartOpen }
            };
            cache.writeData(data);
            return data;
          }
        }
      },
      defaults: {
        cartOpen: false
      }
    }
  });
}

export default withApollo(createClient);
