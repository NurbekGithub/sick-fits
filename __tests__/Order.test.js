import { MockedProvider } from 'react-apollo/test-utils';
import waait from 'waait';
import { mount } from 'enzyme';
import Order, { SINGLE_ORDER_QUERY } from '../components/Order';
import toJSON from 'enzyme-to-json';
import { fakeOrder } from '../lib/testUtils';

const mocks = [
  {
    request: { query: SINGLE_ORDER_QUERY, variables: { id: 'ord123' } },
    result: { data: { order: fakeOrder() } }
  }
]

describe('<Order />', () => {
  it('renders and matches snap', () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Order id='ord123'/>
      </MockedProvider>
    )
    expect(toJSON(wrapper.find('[data-test="orderLoading"]'))).toMatchSnapshot();
  });
})