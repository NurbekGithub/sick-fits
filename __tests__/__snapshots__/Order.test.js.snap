// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> renders and matches snap 1`] = `
<p
  data-test="orderLoading"
>
  Loading...
</p>
`;
